generatebundlefile/ecr.go

```diff
@@ -206,14 +206,33 @@ func (d DockerAuthFile) Remove() error {
 }
 
 // getNameAndVersionPrivate looks up the latest pushed helm chart's tag from a given repo name from ECR.
-func (c *SDKClients) getNameAndVersion(s, accountID string) (string, string, string, error) {
+func (c *SDKClients) getNameAndVersion(repoName, tag, accountID string) (string, string, string, error) {
 	var version string
 	var sha string
-	splitname := strings.Split(s, ":") // TODO add a regex filter
+	splitname := strings.Split(repoName, ":") // TODO add a regex filter
 	name := splitname[0]
-	if len(splitname) == 1 {
+	ecrname := fmt.Sprintf("%s.dkr.ecr.us-west-2.amazonaws.com/%s", accountID, name)
+	if len(splitname) > 0 {
+		// If for promotion, we use a named tag instead of latest we do a lookup for that tag.
+		if !strings.Contains(tag, "latest") {
+			var imagelookup []ecrtypes.ImageIdentifier
+			imagelookup = append(imagelookup, ecrtypes.ImageIdentifier{ImageTag: &tag})
+			ImageDetails, err := c.ecrClient.Describe(&ecr.DescribeImagesInput{
+				RepositoryName: aws.String(repoName),
+				ImageIds:       imagelookup,
+			})
+			if err != nil {
+				return "", "", "", fmt.Errorf("error: Unable to complete DescribeImagesRequest to ECR. %s", err)
```

Comment:
```suggestion
				return "", "", "", fmt.Errorf("DescribeImagesRequest to public ECR failed: %w", err)
```

Error messages shouldn't include the term error, and generally should start with a lower-case letter.

The former is so that a compound error message doesn't have the word error all over the place (plus it's typically added once by the logging package). The latter is more just for formatting, since it's common to add context to an error at multiple levels as it works it way back up the call stack.

Finally, we use `%w` instead of `%s` so that callers can [Unwrap](https://pkg.go.dev/errors#Unwrap) the error if they want, and determine if there was some specific kind of error, or other context in a specific error type.